### PR TITLE
User recommendation/ranking algorithm

### PR DESCRIPTION
**Pull request by Zekereyez**
**Context**
To provide users a more enriched experience we rank the app users against the current user and sort them with an emphasis on experience and their weight class before loading them onto the feed.

**This diff**
Introduced and implemented a ranking algorithm to sort users before loading them onto the feed.

**What’s next**
I will create a complex custom animation for the app.

**TEST PLAN:** 

- Demo: https://imgur.com/a/tD8GxAw
- Here is a consistent log that shows the users being ranked and sorted: 
- 2022-08-02 19:59:01.325167-0700 Squablr[7299:10716857] Name: Ooreoluwa Fasawe, Score: 0.700099
2022-08-02 19:59:01.325461-0700 Squablr[7299:10716857] Name: Craig Lee, Score: 0.700099
2022-08-02 19:59:01.325608-0700 Squablr[7299:10716857] Name: Daphne Lopez, Score: 0.700099
2022-08-02 19:59:01.325767-0700 Squablr[7299:10716857] Name: Nikita Agarwal, Score: 0.700099
2022-08-02 19:59:01.325907-0700 Squablr[7299:10716857] Name: Kleo Ku, Score: 0.700099
2022-08-02 19:59:01.326033-0700 Squablr[7299:10716857] Name: Gregorio Florentino-Sanchez, Score: 0.700099
2022-08-02 19:59:01.326159-0700 Squablr[7299:10716857] Name: Kai Cenat, Score: 0.700187
2022-08-02 19:59:01.326312-0700 Squablr[7299:10716857] Name: Mike Tyson, Score: 0.700245
2022-08-02 19:59:01.326439-0700 Squablr[7299:10716857] Name: Floyd Mayweather, Score: 0.700245
2022-08-02 19:59:01.326559-0700 Squablr[7299:10716857] Name: Blueface, Score: 0.700245
2022-08-02 19:59:01.326667-0700 Squablr[7299:10716857] Name: Baby Keem, Score: 0.700293
2022-08-02 19:59:01.326776-0700 Squablr[7299:10716857] Name: Manny Pacquiao, Score: 0.700480
2022-08-02 19:59:01.326870-0700 Squablr[7299:10716857] Name: GB, Score: 0.700480
2022-08-02 19:59:01.326994-0700 Squablr[7299:10716857] Name: Kendrick Lamar, Score: 0.700480
2022-08-02 19:59:01.327105-0700 Squablr[7299:10716857] Name: Lil Tjay, Score: 0.700480
2022-08-02 19:59:01.327196-0700 Squablr[7299:10716857] Name: Mariana, Score: 0.700521
2022-08-02 19:59:01.327299-0700 Squablr[7299:10716857] Name: Unc Snoop, Score: 0.700521
2022-08-02 19:59:01.327406-0700 Squablr[7299:10716857] Name: NLE Choppa, Score: 0.700620
2022-08-02 19:59:01.338360-0700 Squablr[7299:10716857] Name: A boogie w, Score: 0.700680
2022-08-02 19:59:01.338455-0700 Squablr[7299:10716857] Name: Gervonta Davis, Score: 0.700750
2022-08-02 19:59:01.338592-0700 Squablr[7299:10716857] Name: Nardo Wick, Score: 0.712000
2022-08-02 19:59:01.338718-0700 Squablr[7299:10716857] Name: Central Cee, Score: 0.718750
2022-08-02 19:59:01.338830-0700 Squablr[7299:10716857] Name: Duke, Score: 0.733333
2022-08-02 19:59:01.339064-0700 Squablr[7299:10716857] Name: Muhammad Ali, Score: 1.000000
2022-08-02 19:59:01.339211-0700 Squablr[7299:10716857] Name: Canelo Alvarez, Score: 1.000000
2022-08-02 19:59:01.339321-0700 Squablr[7299:10716857] Name: Tyson Fury, Score: 1.000000
2022-08-02 19:59:01.339425-0700 Squablr[7299:10716857] Name: Pop Smoke, Score: 1.000000
2022-08-02 19:59:01.339538-0700 Squablr[7299:10716857] Name: 21 Savage, Score: 1.000000
2022-08-02 19:59:01.339635-0700 Squablr[7299:10716857] Name: Kodak Black, Score: 1.000000
2022-08-02 19:59:01.339736-0700 Squablr[7299:10716857] Name: NBA Youngboy, Score: 1.000000
2022-08-02 19:59:01.339825-0700 Squablr[7299:10716857] Name: Fivio Foreign, Score: 1.000000
2022-08-02 19:59:01.339914-0700 Squablr[7299:10716857] Name: Tory Lanez, Score: 1.000000
2022-08-02 19:59:01.340016-0700 Squablr[7299:10716857] Name: Eric Moran, Score: 1.000000
2022-08-02 19:59:01.340136-0700 Squablr[7299:10716857] Name: Rami, Score: 1.000000
2022-08-02 19:59:01.340251-0700 Squablr[7299:10716857] Name: ZEKE, Score: 1.000000

### CODE REVIEW

File: Squablr/Models/Profile.m [hunk 1]
```diff
@@ -30,6 +30,7 @@ + (void) writeUserToParse: (PFBooleanResultBlock  _Nullable)completion {
     newUser.stance = @"Orthodox";
     newUser.experience = @(1);
     newUser.biography = @"I'm new here! (:";
+    newUser.userUsageCount = @(0);
```
Comment (rhammoud0418) [inline Squablr/Models/Profile.m]: Get rid of this (I'm guessing this is from the time algorithm thing

File: Squablr/Utils/RankingAlgorithmUtils.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+//8
+
+#import <Foundation/Foundation.h>
+#import "Profile.h"
+#import "Parse/Parse.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RankingAlgorithmUtils : NSObject
++ (void) sortProfilesByCompatibility: (NSMutableArray *) profiles;
```
Comment (rhammoud0418) [inline Squablr/Utils/RankingAlgorithmUtils.h:16]: I like to put a spaces in between the interface and end declaration lines and the body

File: Squablr/Utils/RankingAlgorithmUtils.m [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+}
+
++ (double) _computeCompatibilityScoreForProfile:(Profile *) userProfileFromFeed withCurrentUserProfile:(Profile *) currentUserProfile {
+    double weightScore = [self _weightCalculation:userProfileFromFeed.weightClass withCurrentUserWeight:currentUserProfile.weightClass];
+    double experienceScore = [self _experienceCalculation:userProfileFromFeed.experience withCurrentUserExperience:currentUserProfile.experience];
+    return _WEIGHT_MULTIPLIER * weightScore + _EXPERIENCE_MULTIPLIER * experienceScore;
+}
+
++ (double) _weightCalculation:(NSNumber *) userOnFeedWeight withCurrentUserWeight: (NSNumber *) currentUserWeight;{
+//    NSNumber *difference = currentUserWeight. - currentUserOnFeedWeight;
```
Comment (rhammoud0418) [inline Squablr/Utils/RankingAlgorithmUtils.m]: Commented out code

File: Squablr/View Controllers/UserFeedViewController.m [hunk 4]
```diff
@@ -8,17 +8,21 @@
 #import "UserFeedViewController.h"
 
 @interface UserFeedViewController ()
+@property (nonatomic, strong) NSMutableArray *currentUserProfileInfo;
 @property (nonatomic, strong) NSMutableArray *arrayOfUserObjects;
-@property (nonatomic, strong) NSMutableArray *userProfilePhotos;
+@property (nonatomic, strong) NSMutableArray *userPlaceHolderArray;
```
Comment (rhammoud0418) [inline Squablr/View Controllers/UserFeedViewController.m]: Why do we have all these new properties? Guessing it's leftover from what you were trying to do previously?

**Review by rhammoud0418 — COMMENTED** [review]
Couple comments, other than that LGTM!

File: Squablr/View Controllers/UserFeedViewController.m (hunk 4 above)
Comment (Zekereyez) [reply Squablr/View Controllers/UserFeedViewController.m]: Yes, they are

**Pull request merged**